### PR TITLE
Heretic Nerfs Part 1

### PR DESCRIPTION
**Pull request by Erol509**
## About The Pull Request

Nerfs heretic blade armor penetration by a bit, subject to change. Plus forces them to sacrafice targets which had client linked to them before they died. Making the ascension much harder, and I have plan to make it, that they NEED to sacrafice their further sacraficial targets.

## Why It's Good For The Game

Because they are too powerfull after ascension and before.

## Proof Of Testing

Works so far, still WIF

## Changelog

:cl:
balance: Heretic blade armor pen was lowered from 35 to 25
balance: Heretic Ascension now requires mods with ckey linked to them before they died
/:cl:

### CODE REVIEW

File: code/modules/antagonists/heretic/heretic_knowledge.dm [hunk 1]
```diff
@@ -709,7 +709,7 @@
  * Checks if the passed human is a valid sacrifice for our ritual.
  */
 /datum/heretic_knowledge/ultimate/proc/is_valid_sacrifice(mob/living/carbon/human/sacrifice)
-	return (sacrifice.stat == DEAD) && !ismonkey(sacrifice)
+	return (sacrifice.stat == DEAD) && !ismonkey(sacrifice) && (sacrifice.last_mind) // Bubberstation edit
```
Comment (Majkl-J) [inline code/modules/antagonists/heretic/heretic_knowledge.dm:712]:
```suggestion
	return (sacrifice.stat == DEAD) && !ismonkey(sacrifice) && (!isnull(sacrifice.last_mind)) // BUBBER EDIT - adds last mind requirement
```

**Review by Majkl-J — CHANGES_REQUESTED** [review]
isnull is faster

**Comment by Erol509** [comment]
Changes work so far

**Comment by destrucktoid** [comment]
I think you really need to make it so that if you try to sac someone who is disconnected you instead get that target re-rolled, another idea is to make it so that someone who is disconnected for 10 minutes auto-rerolls. This is because that making it so you cannot sacrifice targets who are disconnected will really screw over your heretic round.

**Comment by Majkl-J** [comment]
> I think you really need to make it so that if you try to sac someone who is disconnected you instead get that target re-rolled, another idea is to make it so that someone who is disconnected for 10 minutes auto-rerolls. This is because that making it so you cannot sacrifice targets who are disconnected will really screw over your heretic round.

last_mind is proofed against this

**Comment by destrucktoid** [comment]
Also, why the nerf to the armour pen? Seems a bit out of the blue and unusual, heretic knives aren't really THAT strong compared to other antagonist weapons so i'm not really sure why it needs a nerf.

**Comment by destrucktoid** [comment]
> > I think you really need to make it so that if you try to sac someone who is disconnected you instead get that target re-rolled, another idea is to make it so that someone who is disconnected for 10 minutes auto-rerolls. This is because that making it so you cannot sacrifice targets who are disconnected will really screw over your heretic round.
> 
> last_mind is proofed against this

What do you mean by it's proofed against this?

**Comment by destrucktoid** [comment]
Wait... I just realised this is making it so that the heretic ASCENSION needs a mind component. This is a big no from me as it will inherently making ascending violate round removal rules.

**Review by projectkepler-ru — CHANGES_REQUESTED** [review]
It'd be better for us to just nerf their spawnrate than the antag itself, the problem is because in higher pop round theres a lot more heretic and more rift for them to loot from.

**Comment by projectkepler-ru** [comment]
Also, Heretic Nerfs Part 1?

**Comment by Erol509** [comment]
> It'd be better for us to just nerf their spawnrate than the antag itself, the problem is because in higher pop round theres a lot more heretic and more rift for them to loot from.

Spawnrate doesnt really matter 90% of the time, even when only 3 spawns, bah 2 spawns. They are able to completly ruin round for everyone on the server, Im thinking about lowpop.

**Comment by Majkl-J** [comment]
> > It'd be better for us to just nerf their spawnrate than the antag itself, the problem is because in higher pop round theres a lot more heretic and more rift for them to loot from.
> 
> Spawnrate doesnt really matter 90% of the time, even when only 3 spawns, bah 2 spawns. They are able to completly ruin round for everyone on the server, Im thinking about lowpop.

Heretics are 40+ pop, at which point at least 4, and a max of 6 can spawn roundstart

This is no only an uninformed assumption, but also applying survivorship bias to the 1/6 that survives long enough to end the round.

**Comment by projectkepler-ru** [comment]
> > > It'd be better for us to just nerf their spawnrate than the antag itself, the problem is because in higher pop round theres a lot more heretic and more rift for them to loot from.
> > 
> > 
> > Spawnrate doesnt really matter 90% of the time, even when only 3 spawns, bah 2 spawns. They are able to completly ruin round for everyone on the server, Im thinking about lowpop.
> 
> Heretics are 40+ pop, at which point at least 4, and a max of 6 can spawn roundstart
> 
> This is no only an uninformed assumption, but also applying survivorship bias to the 1/6 that survives long enough to end the round.


Precisely, most heretic are caught and often time stopped before anything happens, Also, this is not including the fact there are heretics who can cause a lot of damage to the station without ever ascending.

**Comment by KazooBard** [comment]
You do realize ascencion gibs targets, right? This will cause more round removal than you intended to. 

As for the rest, heretics dont ascend often, they are at even odds with sec and the fact they are a threat to every other antag makes sure they are balanced because you're not fighting sec. You're fighting sec and the traitor and changeling and vampire.

**Comment by Erol509** [comment]
> You do realize ascencion gibs targets, right? This will cause more round removal than you intended to.
> 
> As for the rest, heretics dont ascend often, they are at even odds with sec and the fact they are a threat to every other antag makes sure they are balanced because you're not fighting sec. You're fighting sec and the traitor and changeling and vampire.
> 
> EDIT: There is only one person who manages to ascend somewhat consistently and they still die while ascended PRE nerfs, we all know who. Time constraints due to sec chasing etc and even just 4 coordinated officers end a heretic's run quite easily.

There is always a way to remove gib from the heretic ascension ritual.

**Comment by Erol509** [comment]
Guh, I will close it for now. Gonna work on it in the future, I plan on forcing the heretics to fight aginst each other, and sap others for powers. Only one can ascend stepping on the bodies of others.

**Pull request closed without merging**